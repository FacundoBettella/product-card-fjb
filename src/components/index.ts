import { ProductCard  as ProductCardHOC} from "./ProductCard";
import { ProductButtons } from "./ProductButtons";
import { ProductImage } from "./ProductImage";
import { ProductTitle } from "./ProductTitle";

import { IProductCardHOC } from "../interfaces/interfaces";

export { ProductImage } from "./ProductImage";
export { ProductTitle } from "./ProductTitle";
export { ProductButtons } from "./ProductButtons";

export const ProductCard: IProductCardHOC = Object.assign( ProductCardHOC, { 
    Image: ProductImage,
    Title: ProductTitle,
    Buttons: ProductButtons
});

export default ProductCard;


