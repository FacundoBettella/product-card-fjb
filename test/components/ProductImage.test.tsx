import React from "react";
import renderer from "react-test-renderer";
import { ProductImage, ProductCard } from "../../src/components";
import { product2 } from "../data/products";

describe("ProductImage", () => {

    test("Debe mostrar el componente correctamente con la imagen personalizada", () => {
        const wrapper = renderer.create(
            <ProductImage image="./imageSource.jpg" className="productImage" />
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
    });


    test("Debe mostrar el componente con la imagen heredada de su componente padre", () => {
        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {
                    () => (
                        <ProductImage />
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
    })

})