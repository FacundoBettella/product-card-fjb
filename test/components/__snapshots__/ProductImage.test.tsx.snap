// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe mostrar el componente con la imagen heredada de su componente padre 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product img"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage Debe mostrar el componente correctamente con la imagen personalizada 1`] = `
<img
  alt="Product img"
  className="productImg productImage"
  src="./imageSource.jpg"
/>
`;
